app/assets/javascripts/annotations.js

```diff
@@ -745,7 +782,11 @@ function newAnnotationBox(annotation) {
     const problem_id = $(this).val();
 
     // Update autocomplete to display shared comments for selected problem
-    box.find("#comment-textarea").autocomplete("option", "source", localCache["shared_comments"][problem_id] || []);
+    box.find("#comment-textarea").autocomplete({
```

Comment:
Note that it is unnecessary to specify select and focus here. This code is triggered when we change the problem, so we only need to update the source. Same for L526.